### PR TITLE
Bring `shellcheck` into the build process

### PR DESCRIPTION
**Pull request by btoll**
Only check for the presence of the dependencies.  If one or more are missing, exit 1 and echo the install script (`./scripts/configure-dev.sh`).

https://www.shellcheck.net/
https://github.com/koalaman/shellcheck

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,6 +56,11 @@ vet:
 check_license:
 	./scripts/check_license.sh
 
+# Turn on recursive globbing to find all shell scripts.
+check_shell:
+	bash -c "shopt -s globstar"
+	shellcheck */**/*.sh
```
Comment (tsachiherman) [inline Makefile]: I think that you want to do:
```
shopt -s globstar && \
shellcheck */**/*.sh
```
right ?

File: scripts/configure_dev-deps.sh [hunk 2]
```diff
@@ -14,3 +14,23 @@ function install_go_module {
...
+        apt-get update
+        apt-get install sudo -y
+    fi
+
+    sudo apt-get update
+    sudo apt-get install shellcheck -y
+elif [ "$OS" == "Darwin" ]
+then
+    brew install shellcheck
+fi
```
Comment (tsachiherman) [inline scripts/configure_dev-deps.sh]: All that logic need to go to `configure-dev.sh`

Comment (btoll) [reply scripts/configure_dev-deps.sh]: I agree that is where it should go, but I don't think I can move that there without some extra work first.  Here is a grep that shows where else that file is called:

```
~/projects/go-algorand:master$ ag -C 2 configure_dev-deps.sh
docker/build/Dockerfile-deploy
22-    NETWORK=${NETWORK}
23-WORKDIR $GOPATH/src/github.com/algorand/go-algorand
24:RUN scripts/configure_dev-deps.sh && make deps && make clean && find tmp && TMPDIR/deploy_linux_version_exec.sh
25-ENTRYPOINT ["/bin/bash"]
26-

docker/build/Dockerfile
18-    PKG_ROOT=${PKG_ROOT}
19-WORKDIR $GOPATH/src/github.com/algorand/go-algorand
20:RUN scripts/configure_dev-deps.sh && make deps && make clean && scripts/local_install.sh -n
21-ENTRYPOINT ["/bin/bash"]
22-

scripts/build_release_centos_docker.sh
23-REPO_DIR="$( cd "$(dirname "$0")" ; pwd -P )"/..
24-
25:${REPO_DIR}/scripts/configure_dev-deps.sh
26-
27-cd ${REPO_DIR}

scripts/configure_dev.sh
33-fi
34-
35:${SCRIPTPATH}/configure_dev-deps.sh
36-
```

File: Makefile (hunk 1 above)
Comment (btoll) [reply Makefile]: `make` must use the Bourne shell to execute commands:

```
~/projects/go-algorand:shellcheck$ make check_shell 
shopt -s globstar && \
        shellcheck */**/*.sh
/bin/sh: 1: shopt: not found
Makefile:63: recipe for target 'check_shell' failed
make: *** [check_shell] Error 127
```
Can we specify the shell in the `Makefile`?

File: scripts/configure_dev-deps.sh (hunk 2 above)
Comment (tsachiherman) [reply scripts/configure_dev-deps.sh]: -> For the two docker files, you need to add the spellcheck to the `RUN apt` line.
-> for the build_release_centos_docker.sh, it need to be added to the `RUN yum install -y autoconf awscli git gnupg2 nfs-utils python36 sqlite3 boost-devel expect jq libtool gcc-c++ libstdc++-devel libstdc++-static rpmdevtools createrepo rpm-sign bzip2` line.
The last one is good and doesn't need to be modified.

File: Makefile (hunk 1 above)
Comment (tsachiherman) [reply Makefile]: I think that we should attempt to stay more posix compliant here.. 
what's about `find . -name "*.sh" | xargs shellcheck` ?

File: scripts/configure_dev-deps.sh (hunk 2 above)
Comment (btoll) [reply scripts/configure_dev-deps.sh]: Sounds good.

File: Makefile (hunk 1 above)
Comment (btoll) [reply Makefile]: Ok

File: Makefile [hunk 3]
```diff
@@ -56,6 +56,9 @@ vet:
 check_license:
 	./scripts/check_license.sh
 
+check_shell:
+	find . -type f -name *.sh -exec shellcheck {} +
```
Comment (tsachiherman) [inline Makefile]: this is a good idea, but you'll need to have the `*.sh` surrounded by double-quotes. i.e. `"*.sh"`

Comment (btoll) [reply Makefile]: Will change.

**Pull request merged**